src/gasman.c

```diff
@@ -2302,8 +2332,8 @@ UInt CollectBags (
 
     GAP_ASSERT(SanityCheckGasmanPointers());
 
-    /* return success                                                      */
-    return 1;
+    /* success                                                      */
+    return ret;
```

Comment:
With https://github.com/gap-system/gap/pull/3797 you'd only have to remove the `if (done == 2)` check, and instead change the return here to `return (done == 1)` or `return done != 2` or something like that. And no need for `ret`, and most of the other changes in `CollectBags` (except for the change to that one `SyAllocBags` call, of course)